lib/inertia_rails/renderer.rb

```diff
@@ -16,7 +16,11 @@ def initialize(component, controller, request, response, render_method, props: n
     end
 
     def render
-      @response.set_header('Vary', [@request.headers['Vary'], 'X-Inertia'].compact.join(', '))
+      if @response.headers["Vary"].blank?
```

Comment:
To be honest, I prefer the original functional approach, but it generates unnecessary objects, which can affect production performance due to additional GC load.